### ISSUE TITLE
how to use model only with my custom dataset and custom training framework

### ISSUE THREAD

**Issue by mes12310** [open-mmlab/mmsegmentation#2971]
Hello. I am new to MMSegmentation.

I am working on a semantic segmentation project where I am comparing multiple networks. I have implemented my own dataset loading and training method using PyTorch only. One model I want to look at is Segmenter, which MMSegmentation has. So, I **only** want to load the model without using any other MMSegmention module like mmEngine.   Also, I want to train it once from scratch and once using pretrained imagenet ViT weights for the encoder. However, I am getting confused on how to only load the model without any other mmseg modules. My images are 224x224. To get the model (Seg-S-Mask/16) , I am trying the following: 

```
model = dict(
            type='EncoderDecoder',
            backbone=dict(
                type='VisionTransformer',
                img_size=(224, 224),
                patch_size=16,
                in_channels=3,
                embed_dims=384,
                num_layers=12,
                num_heads=6,
                drop_path_rate=0.1,
                attn_drop_rate=0.0,
                drop_rate=0.0,
                final_norm=True,
                with_cls_token=True,
                interpolate_mode='bicubic',
            ),
            decode_head=dict(
                type='SegmenterMaskTransformerHead',
                in_channels=384,
                channels=384,
                num_classes=4,
                num_layers=2,
                num_heads=6,
                embed_dims=384,
                dropout_ratio=0.0,
                loss_decode=dict(
                    type='CrossEntropyLoss', use_sigmoid=False, loss_weight=1.0),
            )
        )
        mdl_config = Config(dict(model=model))
        model = build_segmentor(mdl_config.model)
```
However, the model output is the wrong shape 
```
Layer (type:depth-idx)                             Input Shape               Output Shape              Param #
EncoderDecoder                                     [32, 3, 224, 224]         [32, 4, 14, 14]           --
```

1. How do I fix this?
2. How do i modify this to train from scratch and using imagenet encoder weights?
